component/src/main/java/org/wso2/extension/siddhi/store/mongodb/MongoDBEventTable.java

```diff
@@ -240,30 +286,24 @@ protected void init(TableDefinition tableDefinition, ConfigReader configReader)
 
         this.initializeConnectionParameters(storeAnnotation, configReader);
 
-        List<IndexModel> expectedIndexModels = new ArrayList<>();
+        this.expectedIndexModels = new ArrayList<>();
         IndexModel primaryKey = MongoTableUtils.extractPrimaryKey(primaryKeys, this.attributeNames);
         if (primaryKey != null) {
-            expectedIndexModels.add(primaryKey);
+            this.expectedIndexModels.add(primaryKey);
         }
-        expectedIndexModels.addAll(MongoTableUtils.extractIndexModels(indices, this.attributeNames));
+        this.expectedIndexModels.addAll(MongoTableUtils.extractIndexModels(indices, this.attributeNames));
 
         String customCollectionName = storeAnnotation.getElement(
                 MongoTableConstants.ANNOTATION_ELEMENT_COLLECTION_NAME);
         this.collectionName = MongoTableUtils.isEmpty(customCollectionName) ?
                 tableDefinition.getId() : customCollectionName;
-
-        if (!this.collectionExists()) {
-            try {
-                this.getDatabaseObject().createCollection(this.collectionName);
-                this.createIndices(expectedIndexModels);
-            } catch (MongoException e) {
-                this.mongoClient.close();
-                throw new MongoTableException("Creating mongo collection '" + this.collectionName
-                        + "' is not successful due to " + e.getLocalizedMessage(), e);
-            }
-        } else {
-            MongoCursor<Document> existingIndicesIterator = this.getCollectionObject().listIndexes().iterator();
-            MongoTableUtils.checkExistingIndices(expectedIndexModels, existingIndicesIterator);
+        this.initialCollectionTest = false;
+        try {
+            this.mongoClient = new MongoClient(this.mongoClientURI);
+        } catch (MongoException e) {
+            throw new MongoTableException("Annotation 'Store' contains illegal value for " +
```

Comment:
Why are we throwing this runtime exception to Siddhi? If you want initialization to fail shall we throw SiddhiAppCreationException? WDYT?